mlkem/asm/aarch64/rej_uniform_asm_clean.S

```diff
@@ -0,0 +1,585 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*************************************************
+* Name:        rej_uniform_asm
+*
+* Description: Full SIMD lane, run rejection sampling on uniform random bytes to generate
+*              uniform random integers mod q
+*
+* Arguments:   - int16_t *r:          pointer to output buffer
+*              - unsigned int len:    requested number of 16-bit integers
+*                                     (uniform mod q)
+*              - const uint8_t *buf:  pointer to input buffer
+*                                     (assumed to be uniform random bytes)
+*              - unsigned int buflen: length of input buffer in bytes
+*              - unsigned int *consumed: the length of consumed buffer
+* Returns number of sampled 16-bit integers (at most len)
+***************************************************/
+
+#include "config.h"
+#if defined(MLKEM_USE_AARCH64_ASM)
+
+// Needed to provide ASM_LOAD directive
+#include "common.i"
+
+/* Return multiple of 48: (n/48) * 48 */
+.macro round_48x    dst, src
+    mov     w8, #48
+    udiv    \dst, \src, w8
+    mul     \dst, \dst, w8
```

Comment:
I'm not a fan of this at all.

Reply:
This is the division by constant, where is `n` is the `buflen` input.

Reply:
Sure, everything is public in the rejection sampling, but this is not the fastest way to compute division by 48.
Additionally, the current code cannot handle anything that's not a multiple of 24, so there is really no reason to do that division here.

Reply:
I removed this in the latest commit